packages/flutter/test/widgets/async_test.dart

```diff
@@ -81,6 +94,22 @@ void main() {
       await eventFiring(tester);
       expect(find.text('AsyncSnapshot(ConnectionState.none, null, null)'), findsOneWidget);
     });
+    testWidgets('gracefully handles transition to null future with initial data without introducing new states', (WidgetTester tester) async {
```

Comment:
About the test name: What is actually tested is that when reconfiguring a `FutureBuilder`, the `initialData` is ignored. So I suggest renaming the test to `ignores initialData when reconfiguring`.

I think we should avoid retesting that the `FutureBuilder` gracefully handles its transitions. The `initialData` feature is orthogonal to that. That should cut down on the number of test cases below.